x-pack/test/functional/services/machine_learning/api.ts

```diff
@@ -377,6 +408,36 @@ export function MachineLearningAPIProvider({ getService }: FtrProviderContext) {
       });
     },
 
+    async assertDataFrameAnalyticsJobExist(analyticsId: string) {
+      await retry.waitForWithTimeout(
+        `results for job ${analyticsId} to exist`,
+        30 * 1000,
+        async () => {
+          const job = await esSupertest.get(`/_ml/data_frame/analytics/${analyticsId}`);
+          if (job.status === 200) {
+            return true;
+          } else {
+            throw new Error(`expected data frame analytics job '${analyticsId}' to exist`);
+          }
+        }
+      );
+    },
+
+    async assertDataFrameAnalyticsJobNotExist(analyticsId: string) {
```

Comment:
I would suggest to have this implemented similar to `waitForDataFrameAnalyticsJobToExist` and make use of your `getDataFrameAnalyticsJob` enhancement. Something like
```
    async waitForDataFrameAnalyticsJobNotToExist(analyticsId: string) {
      await retry.waitForWithTimeout(`'${analyticsId}' not to exist`, 5 * 1000, async () => {
        if (await this.getDataFrameAnalyticsJob(analyticsId, 404)) {
          return true;
        } else {
          throw new Error(`expected data frame analytics job '${analyticsId}' not to exist`);
        }
      });
    },
```